### DISCUSSION TITLE
Class-IL Permuted MNIST

### DISCUSSION THREAD

**Discussion by en-tropyc** [ContinualAI/avalanche#609] [Q&A] [answered]
I'm trying to set up a class incremental learning scenario using the Permuted MNIST task protocol. I modified the PermutedMNIST() method from [cmnist.py](https://github.com/ContinualAI/avalanche/blob/master/avalanche/benchmarks/classic/cmnist.py) so that an `NCScenario` instance is returned as follows:

    return nc_benchmark(
        list_train_dataset, 
        list_test_dataset, 
        n_experiences=len(list_train_dataset), 
        shuffle=False, 
        task_labels=False,
        class_ids_from_zero_from_first_exp=False,
        class_ids_from_zero_in_each_exp=False,
        one_dataset_per_exp=True
    )

Below are some of the sampled image/label pairs from each of the resulting experiences. This looks correct to me since at test time the network will only see `Task 0` for each different permutation it's confronted with, but it will need to correctly infer the digit and infer permutation it belongs to (ex. digit 7 from permutation 2 for output 27). Am I thinking about this correctly?

![image](https://user-images.githubusercontent.com/20301223/118062234-68f8ff80-b34b-11eb-9891-da47a75d25c9.png)

Thank you in advance.

**Comment by AndreaCossu** [comment, marked as answer]
Hi @christam96 ! Yes, by looking at the labels tensor you have the result seems to be correct. You are in a class-incremental setting, just set the output layer of your model to have the appropriate number of units and you are good to go.

**Reply by en-tropyc** [reply]
Will do, thanks for the help 😃
